feature: Schedule Tweaks

LeonelCeballos18 opened the pull request:
## What kind of PR is this?
- [x] Feature

## Description
-add `schedule` entity
-add `schedule` entity in data source
-add `create and update dto` for schedule methods
-add crud in `schedule service`
-add `schedule controller`
-add `schedule service and controller` test
-add `schedule dtos` in index
-add `schedule module` in app module
-add `schedule module` in index

## Ticket id
https://barbershop-app.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-78

### Test
![image](https://github.com/user-attachments/assets/b31b2eed-bcd2-46ea-909a-2884f8984c22)
![image](https://github.com/user-attachments/assets/eeaef1d0-74fa-4947-98b8-38d8aa15ffea)

LeonelCeballos18 committed:
feature: Schedule Tweaks

-add `schedule` entity
-add `schedule` entity in data source
-add `create and update dto` for schedule methods
-add crud in `schedule service`
-add `schedule controller`
-add `schedule service and controller` test
-add `schedule dtos` in index
-add `schedule module` in app module
-add `schedule module` in index

cesarj45 reviewed (approved):
LGTM

cmagallon1 commented on src/controllers/schedule.controller.spec.ts:
```diff
@@ -0,0 +1,115 @@
+    save: jest.fn(),
+    remove: jest.fn(),
+    create: jest.fn(),
+    findOneBy: jest.fn(), // Asegúrate de simular 'findOneBy'
```
And did you successfully simulated the findOneBy? 👀

Jorge-Antonio-Torres-Lozoya replied:
😂

cmagallon1 commented:
One question, what's the purpose of this schedule controller? 🤔

cmagallon1 commented on src/services/schedue.service.ts:
```diff
@@ -0,0 +1,69 @@
+
+  async create(dto: CreateScheduleDTO[]): Promise<Schedule[]> {
+    for (const scheduleDto of dto) {
+      const existingSchedule = await this.scheduleRepository.findOne({
```
I want to highlight, which this change you won't be able to create records with the same day, that means you can only create 7 records (the seven days of the week) is that intended?

If so, I'd say to create a unique index then in the schedule table for the day, so it will be validated at the db level too

LeonelCeballos18 replied:
Just solved

cmagallon1 replied:
is more of a question actually, was that intended?, do you wanted to store a record per day of the week?

LeonelCeballos18 replied:
It supposed to storage only the six days that kevin works, and update his schedule if is necessary

cmagallon1 replied:
okay, got it

I'm asking because the way I see it, this can be a seed. Since these timings are fixed, you can add the seven days of the week in a seed along with the open_hours. This way, you don't need to create that info in each environment, including your local machine

Can you add these defined open hours to the seed file, please?

LeonelCeballos18 replied:
yes, sounds better, but I think that it's ok if I keep the update and delete method, right?

cmagallon1 replied:
right!

LeonelCeballos18 committed:
Solving comments

-add unique index in day field and migration
-add `AuthGuard` in patch controller

cmagallon1 commented:
> One question, what's the purpose of this schedule controller? 🤔

@LeonelCeballos18

LeonelCeballos18 commented:
> > One question, what's the purpose of this schedule controller? 🤔
> 
> @LeonelCeballos18

It's because in frontend we have the schedule in an array, but in the administrator panel kevin can modify his own schedule, so that's why

LeonelCeballos18 committed:
Solving comments

-add schedule seed
-remove create method of schedule
